src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -317,20 +388,34 @@ private void ModifyDotNetExePathInWebConfig()
         // Transforms the web.config file to set attributes like hostingModel="inprocess" element
         private void ModifyAspNetCoreSectionInWebConfig(string key, string value)
         {
-            var webConfigFile = $"{DeploymentParameters.PublishedApplicationRootPath}/web.config";
-            var config = XDocument.Load(webConfigFile);
+            var config = XDocument.Load(_webConfig);
             var element = config.Descendants("aspNetCore").FirstOrDefault();
             element.SetAttributeValue(key, value);
-            config.Save(webConfigFile);
+            config.Save(_webConfig);
         }
 
         private void ModifyHandlerSectionInWebConfig(string key, string value)
         {
-            var webConfigFile = $"{DeploymentParameters.PublishedApplicationRootPath}/web.config";
-            var config = XDocument.Load(webConfigFile);
+            var config = XDocument.Load(_webConfig);
             var element = config.Descendants("handlers").FirstOrDefault().Descendants("add").FirstOrDefault();
             element.SetAttributeValue(key, value);
-            config.Save(webConfigFile);        
+            config.Save(_webConfig);
+        }
+
+        // These elements are duplicated in the web.config if you publish. Remove them from the host.config.
+        private string RemoveRedundantElements(string serverConfig)
```

Comment:
Doesn't IIS merge them?

Reply:
No, it refuses to load the site if you have duplicates.